eval.go

```diff
@@ -302,25 +306,31 @@ func (m sortByOffsetSlice) Less(i, j int) bool {
 func gatherMatches(mt matchTree, known map[matchTree]bool) []*candidateMatch {
 	var cands []*candidateMatch
 	visitMatches(mt, known, func(mt matchTree) {
-		if smt, ok := mt.(*substrMatchTree); ok {
-			cands = append(cands, smt.current...)
-		}
-		if rmt, ok := mt.(*regexpMatchTree); ok {
-			cands = append(cands, rmt.found...)
+		switch t := mt.(type) {
+		case *substrMatchTree:
+			cands = append(cands, t.current...)
+		case *regexpMatchTree:
+			cands = append(cands, t.found...)
+		case *symbolMatchTree:
+			cands = append(cands, t.cands...)
 		}
 	})
 
-	foundContentMatch := false
+	scope := query.ScopeFileName
 	for _, c := range cands {
-		if !c.fileName {
-			foundContentMatch = true
+		switch c.scope {
+		case query.ScopeFileContent, query.ScopeSymbol:
+			scope = c.scope
```

Comment:
shouldn't you just break here? Do you want to include symbol and content matches together? If so below (line 333) will only include one.